app/src/androidTest/java/org/mozilla/focus/activity/PrivateBrowsingTest.java

```diff
@@ -0,0 +1,134 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 20; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.activity;
+
+import android.Manifest;
+import android.content.Intent;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.rule.GrantPermissionRule;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mozilla.focus.Inject;
+import org.mozilla.focus.R;
+import org.mozilla.focus.persistence.TabEntity;
+import org.mozilla.focus.persistence.TabsDatabase;
+import org.mozilla.focus.utils.AndroidTestUtils;
+import org.mozilla.focus.web.GeoPermissionCache;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.Espresso.pressBack;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDescendantOfA;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.isSelected;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.core.AllOf.allOf;
+import static org.mozilla.focus.utils.RecyclerViewTestUtils.atPosition;
+
+@RunWith(AndroidJUnit4.class)
+public class PrivateBrowsingTest {
+
+    private static final TabEntity TAB = new TabEntity("TEST_ID", "ID_HOME", "Yahoo TW", "https://tw.yahoo.com/");
```

Comment:
can we use a local file like GPL.html? So we don't need permission. And also the test will run faster.